apis/condition_set.go

```diff
@@ -334,7 +334,7 @@ func (r conditionsImpl) findUnhappyDependent() *Condition {
 	}
 
 	// If something was not initialized.
-	if len(r.dependents) != len(conditions) {
+	if len(r.dependents) > len(conditions) {
```

Comment:
I don't understand the purpose of this change, perhaps a unit test to illustrate the problem case?